stetho/src/main/java/com/facebook/stetho/Stetho.java

```diff
@@ -132,11 +132,18 @@ public static InspectorModulesProvider defaultInspectorModulesProvider(final Con
       @Override
       public Iterable<ChromeDevtoolsDomain> get() {
         ArrayList<ChromeDevtoolsDomain> modules = new ArrayList<ChromeDevtoolsDomain>();
-        modules.add(new Console());
         modules.add(new CSS());
         modules.add(new Debugger());
+        Runtime runtime = new Runtime();
+        modules.add(runtime);
         if (Build.VERSION.SDK_INT >= AndroidDOMConstants.MIN_API_LEVEL) {
-          modules.add(new DOM(new AndroidDOMProviderFactory((Application)context.getApplicationContext())));
+          DOM dom = new DOM(
+              new AndroidDOMProviderFactory(
+                  (Application)context.getApplicationContext()));
+          modules.add(dom);
+          modules.add(new Console(dom, runtime));
```

Comment:
So now the dependency from `Console` to `DOM` and `Runtime` is part of the public contract? You say you want to avoid frequent and possibly confusing changes to the API, but I think that's precisely what you're introducing here. Why do I need to create `DOM` and `Runtime` and then plug those into `Console`? What happens when we change our mind and `Console` no longer needs either of these? Do I need to supply the same `DOM` to `Console` that I already plugged into a call to `modules.add()`, or should I add a new one? This all sounds pretty confusing and arbitrary to me.

I think it'd be better to handle this internally somehow. For example, `DescriptorMap` does initialization of `Descriptor`s in 2 phases. First you register `Descriptor`s, and then in phase 2 they all get linked together by `Descriptormap`.

So maybe `ChromeDevtoolsDomain` could have two new methods, `Class<?>[] getDependencies()` and `void provideDependency(Class<?>, Object)`. `Console`'s implementation of `getDependencies` would return `{ DOM.class, Runtime.class }` and be given the appropriate stuff in `provideDependencies`.

It's hardly a wonderful or elegant solution *but* it keeps this dependency management as an internal Stetho problem.